users/users-model.js

```diff
@@ -0,0 +1,34 @@
+const db = require('../data/dbConfig');
+
+module.exports = {
+  find,
+  findById,
+  add,
+  findBy
+}
+
+function find() {
+  return db('users').select('id', 'username', 'password');
+}
+
+function findBy(filter){
+  return db('users').where(filter);
+}
+
+function findById(id){
+  return db('users')
+    .select('id', 'username', 'password')
+    .where({ id })
+    .first();
+}
+// POST - /api/register - Creates a `user` using the information sent inside the `body` of the request. **Hash the password** before saving the user to the database. 
+async function add(user){
+  const [id] = await db('users').insert(user);
+  return findById(id);
```

Comment:
You could use arrow functions for all of these! For example, an `async` arrow function:
```javascript
const add = async user => {
  const [ id ] = await db('users').insert(user);
  return findById(id);
};
```

and, your `findBy` function for an example of an implicit return:
```javascript
const findBy = filter => db('users').where(filter);
```